core/src/main/java/com/novoda/merlin/service/HostPinger.java

```diff
@@ -20,12 +23,12 @@
 
     public static HostPinger newInstance(PingerCallback pingerCallback) {
         MerlinLog.d("Host address not set, using Merlin default : " + Merlin.DEFAULT_ENDPOINT);
-        return newInstance(pingerCallback, Merlin.DEFAULT_ENDPOINT);
+        PingTaskFactory pingTaskFactory = new PingTaskFactory(pingerCallback, new ResponseCodeFetcher(), new DefaultEndpointResponseCodeValidator());
+        return new HostPinger(pingerCallback, Merlin.DEFAULT_ENDPOINT, pingTaskFactory);
     }
 
     public static HostPinger newInstance(PingerCallback pingerCallback, String hostAddress) {
-        ResponseCodeFetcher responseCodeFetcher = new ResponseCodeFetcher();
-        PingTaskFactory pingTaskFactory = new PingTaskFactory(pingerCallback, responseCodeFetcher);
+        PingTaskFactory pingTaskFactory = new PingTaskFactory(pingerCallback, new ResponseCodeFetcher(), new CustomEndpointResponseCodeValidator());
```

Comment:
you can have yet another factory method then, it's a bit sad you are tied to the default endpoint for the feature, what all you need is knowing in advance what'll be returned